compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1060,8 +1060,8 @@ impl<'a> Parser<'a> {
             })
         } else if self.eat_keyword(kw::Unsafe) {
             self.parse_block_expr(None, lo, BlockCheckMode::Unsafe(ast::UserProvided), attrs)
-        } else if self.check_inline_const() {
-            self.parse_const_expr(lo.to(self.token.span))
+        } else if self.eat_keyword(kw::Const) {
```

Comment:
I think @petrochenkov and others wanted to avoid eating the keyword before checking if it's really an inline const. My first reaction was like ... but we're eating `Unsafe` right above. I'm not sure if there's another way to solve this problem and we could still rely on checking or if we just need to eat the keyword.

Anyway, would leave this to @petrochenkov

Reply:
The difference is that `unsafe` just modifies the block expression, whereas `const` needs to create the `ExprKind::Const` - but maybe it doesn't have to be too complicated *shrug*.